site/src/component/GradeDist/GradeDist.tsx

```diff
@@ -156,7 +156,7 @@ const GradeDist: FC<GradeDistProps> = (props) => {
 
   // update list of quarters when new professor/course is chosen
   useEffect(() => {
-    if ((currentProf || currentCourse) && gradeDistData.length !== 0) {
+    if ((currentProf || currentCourse) && gradeDistData && gradeDistData.length !== 0) {
```

Comment:
If you want it to be more concise, you can do:
```suggestion
    if ((currentProf || currentCourse) && gradeDistData?.length) {
```

That's readable to me, but I may be used to it more than others. Your call.